docs/content/reference/konnectivity.md

```diff
@@ -0,0 +1,393 @@
+names, so it can only be tested with IP addresses.
+
+```sh
+#!/bin/bash
+
+set -euo pipefail
+
+workdir="$(mktemp -d)"
+cp_namespace="CONTROL_PLANE_NAMESPACE_HERE"
+
+echo "work directory is: ${workdir}"
+
+# Get the cert/CA required to use the konnectivity server as a proxy
+oc get secret konnectivity-client -n ${cp_namespace} -o jsonpath='{ .data.tls\.key }' | base64 -d > "${workdir}/client.key"
+oc get secret konnectivity-client -n ${cp_namespace} -o jsonpath='{ .data.tls\.crt }' | base64 -d > "${workdir}/client.crt"
+oc get cm konnectivity-ca-bundle -n ${cp_namespace} -o jsonpath='{ .data.ca\.crt }' > "${workdir}/konnectivity_ca.crt"
+
+# Get the cert/CA required to access the kubelet endpoint
+oc get cm kubelet-client-ca -n ${cp_namespace} -o jsonpath='{ .data.ca\.crt }' > ${workdir}/kubelet_ca.crt
+oc get secret kas-kubelet-client-crt -n ${cp_namespace} -o jsonpath='{ .data.tls\.crt }' | base64 -d > ${workdir}/kubelet_client.crt
+oc get secret kas-kubelet-client-crt -n ${cp_namespace} -o jsonpath='{ .data.tls\.key }' | base64 -d > ${workdir}/kubelet_client.key
+
+# Obtain a node IP from local machines
+nodeip="$(oc get machines -n ${cp_namespace} -o json | jq -r '.items[0].status.addresses[] | select(.type=="InternalIP") | .address')"
+
+# Forward the konnectivity server endpdoint to the local machine
+oc port-forward -n ${cp_namespace} svc/konnectivity-server-local 8090:8090 &
+
+# Allow some time for the port-forwarding to start
+sleep 2
+
+# Perform the curl command with the localhost konnectivity endpoint
+curl -x "https://127.0.0.1:8090" \
+  --proxy-cacert ${workdir}/konnectivity_ca.crt \
+  --proxy-cert ${workdir}/client.crt \
+  --proxy-key ${workdir}/client.key \
+  --cacert ${workdir}/kubelet_ca.crt \
+  --cert ${workdir}/kubelet_client.crt \
+  --key ${workdir}/kubelet_client.key \
+  "https://\${nodeip}:10250/metrics"
+
+# Kill the port-forward job
+kill %1
+```
+
+## Konnectivity Q&A
+
+**Upstream allows grpc or http-connect mode for the konnectivity protocol, why does HyperShift use http-connect?**
+We need to use http-connect to allow clients other than the Kubernetes
+API server to send traffic through the konnectivity server. The socks5
+and http proxies can use http-connect to establish a connection through
+konnectivity-server.
+
+**Upstream allows connecting to the konnectivity server via unix domain socket or via tcp port. Why does HyperShift use tcp port?**
+The konnectivity server only allows one way of serving (via UDS or TCP
+port). We need to use a TCP port so we can expose it as a service to
+clients that use the socks5 and http proxies.
+
+**Why does Konnectivity server run as a sidecar of the Kubernetes API server and not in its own deployment?**
+When running as a sidecar, the Kubernetes API server can communicate
+with the Konnectivity server very fast, since traffic only goes through
+the same pod's interface. It also allows the Kubernetes API server to
```

Comment:
s/only goes through the same pod's interface/happens over loopback/

is this right? i thought we used loopback for this in the kas egress config.

Reply:
it's the same interface since it's the same network namespace, but yes loopback is clearer.